backend/src/graphql/resolver/util/creations.test.ts

```diff
@@ -0,0 +1,268 @@
+            await expect(getUserCreation(user.id, 0)).resolves.toEqual([
+              expect.decimalEqual(550),
+              expect.decimalEqual(340),
+              expect.decimalEqual(350),
+            ])
+          })
+        })
+
+        describe('call getUserCreation with JST (GMT+0900)', () => {
+          it('returns the expected open contributions', async () => {
+            await expect(getUserCreation(user.id, -540, true)).resolves.toEqual([
+              expect.decimalEqual(340),
+              expect.decimalEqual(350),
+              expect.decimalEqual(1000),
+            ])
+          })
+        })
+
+        describe('call getUserCreation with PST (GMT-0800)', () => {
+          it('returns the expected open contributions', async () => {
+            await expect(getUserCreation(user.id, 480, true)).resolves.toEqual([
+              expect.decimalEqual(550),
+              expect.decimalEqual(340),
+              expect.decimalEqual(350),
+            ])
+          })
+        })
+
+        describe('run two hours forward to be in the next month in UTC', () => {
+          const nextMonthTargetDate = new Date()
+          nextMonthTargetDate.setTime(targetDate.getTime() + 2 * 60 * 60 * 1000)
+
+          beforeAll(() => {
+            /* eslint-disable-next-line @typescript-eslint/no-empty-function */
+            setTimeout(() => {}, 2 * 60 * 60 * 1000)
+            jest.runAllTimers()
+          })
+
+          it('has the clock set correctly', () => {
+            expect(new Date().toISOString()).toContain(
+              `${nextMonthTargetDate.getFullYear()}-${nextMonthTargetDate.getMonth() + 1}-01T01:`,
+            )
+          })
+
+          describe('call getUserCreation with UTC', () => {
+            it('returns the expected open contributions', async () => {
+              await expect(getUserCreation(user.id, -540, true)).resolves.toEqual([
+                expect.decimalEqual(340),
+                expect.decimalEqual(350),
+                expect.decimalEqual(1000),
+              ])
+            })
+          })
+
+          describe('call getUserCreation with JST (GMT+0900)', () => {
+            it('returns the expected open contributions', async () => {
+              await expect(getUserCreation(user.id, -540, true)).resolves.toEqual([
+                expect.decimalEqual(340),
+                expect.decimalEqual(350),
+                expect.decimalEqual(1000),
+              ])
+            })
+          })
+
+          describe('call getUserCreation with PST (GMT-0800)', () => {
+            it('returns the expected open contributions', async () => {
+              await expect(getUserCreation(user.id, 450, true)).resolves.toEqual([
+                expect.decimalEqual(550),
+                expect.decimalEqual(340),
+                expect.decimalEqual(350),
+              ])
+            })
+          })
+        })
+      })
+    })
+  })
+})
```

Comment:
Very good and detailed tests!!!
Congratulations...

Reply:
Thanks.